xbmc/platform/linux/peripherals/PeripheralBusUSBLibUdev.cpp

```diff
@@ -72,33 +72,25 @@ CPeripheralBusUSB::CPeripheralBusUSB(CPeripherals& manager) :
 
   m_udev          = NULL;
   m_udevMon       = NULL;
-
-  if (!(m_udev = udev_new()))
-  {
-    CLog::Log(LOGERROR, "%s - failed to allocate udev context", __FUNCTION__);
-    return;
-  }
-
-  /* set up a devices monitor that listen for any device change */
-  m_udevMon = udev_monitor_new_from_netlink(m_udev, "udev");
-  udev_monitor_enable_receiving(m_udevMon);
-
-  CLog::Log(LOGDEBUG, "%s - initialised udev monitor", __FUNCTION__);
 }
 
 CPeripheralBusUSB::~CPeripheralBusUSB(void)
 {
   StopThread(true);
-  udev_monitor_unref(m_udevMon);
-  udev_unref(m_udev);
 }
 
 bool CPeripheralBusUSB::PerformDeviceScan(PeripheralScanResults &results)
 {
   struct udev_enumerate *enumerate;
   struct udev_list_entry *devices, *dev_list_entry;
   struct udev_device *dev(NULL), *parent(NULL);
-  enumerate = udev_enumerate_new(m_udev);
+  struct udev* udev_scan;
+  if (!(udev_scan = udev_new()))
```

Comment:
this makes a single thread having two udev contexts at the same time, if called from internal thread. imo this can be done better.  a thread from outside should not perform a direct scan, instead trigger a scan on the internal thread.

Reply:
You are right. Let's see when I get a minute to do it. Cannot promise anything.